query/src/pipelines/new/executor/executor_tasks.rs

```diff
@@ -0,0 +1,230 @@
+use std::collections::VecDeque;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::sync::{Arc, Condvar};
+use std::thread::Thread;
+use futures::future::BoxFuture;
+use petgraph::prelude::NodeIndex;
+
+use common_exception::Result;
+use common_infallible::Mutex;
+
+use crate::pipelines::new::executor::executor_worker_context::{ExecutorWorkerContext, ExecutorTask};
+use crate::pipelines::new::processors::processor::ProcessorPtr;
+
+pub struct ExecutorTasksQueue {
+    finished: AtomicBool,
+    workers_tasks: Mutex<ExecutorTasks>,
+}
+
+impl ExecutorTasksQueue {
+    pub fn create(workers_size: usize) -> ExecutorTasksQueue {
+        ExecutorTasksQueue {
+            finished: AtomicBool::new(false),
+            workers_tasks: Mutex::new(ExecutorTasks::create(workers_size)),
+        }
+    }
+
+    pub fn finish(&self) {
+        self.finished.store(true, Ordering::Relaxed);
+    }
+
+    pub fn is_finished(&self) -> bool {
+        self.finished.load(Ordering::Relaxed)
+    }
+
+    // Pull task from the global task queue
+    pub fn steal_task_to_context(&self, context: &mut ExecutorWorkerContext) {
+        unsafe {
```

Comment:
It's better to unsafe the block who needs the unsafe.
And sign a SAFETY to explaining, for example:
https://github.com/Rust-for-Linux/linux/commit/632ef868c289eebddfb2d8e60b2c11aad578e5e0#diff-5a3db4f13149a3527a4b10c0fbc104533afb9bdb77878fa29b0027e198d0c22cR62